### ISSUE TITLE
508-defect-2 [SEMANTICS]: Radio button component's ariaDescribedby attribute must reference a valid ID

### ISSUE THREAD

**Issue by 1Copenut** [department-of-veterans-affairs/va.gov-team#28404]
# [508-defect-2](https://github.com/department-of-veterans-affairs/va.gov-team/blob/master/platform/accessibility/guidance/defect-severity-rubric.md#508-defect-2)

## Feedback framework

- **❗️ Must** for if the feedback must be applied 
- **⚠️ Should** if the feedback is best practice 
- **✔️ Consider** for suggestions/enhancements 

## Definition of done

1. Review and acknowledge feedback.
1. Fix and/or document decisions made.
1. Accessibility specialist will close ticket after reviewing documented decisions / validating fix.

## Point of Contact

**VFS Point of Contact:** _Trevor_

## Details

Our [RadioButton component](https://design.va.gov/storybook/?path=/docs/components-radiobuttons--aria-describedby) that uses the `ariaDescribedby` prop is set to an array indexed string, but not necessarily a valid ID on the page. This creates a situation where the aria-describedby text is not read aloud when we show additional content when the radio button is checked.

I've mocked up a Codepen ~~(currently being tested by a11y specialists)~~ to show how this should behave with regard to adding or removing the `aria-describedby` attribute: https://codepen.io/tpierce_402/full/BaRqOzR . This pen has been evaluated by the accessibility specialists and given the 👍 

## Acceptance Criteria

- [ ] `aria-describedby` attribute appears when additional content is revealed (on radio button checked)
- [ ] Screen readers announce the `aria-describedby` properly
- [ ] IDs are unique for the shown/hidden content
- [ ] Zero axe-core violations

## WCAG or Vendor Guidance (optional)

* https://www.w3.org/WAI/WCAG21/Understanding/name-role-value.html

## Screenshots or Trace Logs

![Screen Shot 2021-08-06 at 10 21 36 AM](https://user-images.githubusercontent.com/934879/128546051-26ad0eb5-9e5a-47e7-9ab8-82e0188fecbf.png)

**Comment by caw310** [comment]
@pdavies88 @andresriveratoro @bkjohnson @k80bowman , Would one of you provide guidance on updating to the web component that would address the issue in this ticket?

**Comment by k80bowman** [comment]
For context, when I looked into this issue briefly, it seems like this is something that could now be done programmatically in the application rather than introducing this logic inside the component. But it would be good to confirm this.

**Comment by bkjohnson** [comment]
I think I may have found a bug in the web component where selecting an option doesn't apply `aria-describedby` even though it should.

With that aside, I'm reading this as a storybook update, not anything in particular that app teams need to do. However, the web component doesn't have that "expanded content" feature that the react version does (and that's what Trevor is using for the `aria-describedby` id, so we may want to add that feature. I believe the decision was to build that out as/when needed.
